roles/rsyslog/tasks/main.yml

```diff
@@ -55,14 +55,14 @@
       file:
         state: directory
         path: '{{ __rsyslog_config_dir }}'
-        mode: '0755'
+        mode: preserve
```

Comment:
I think this should be explicitly set to a mode - so `mode: '0755'` is ok

You should only use `mode: preserve` when you are absolutely sure that you want to use the file mode from the source for the destination.  In almost every case here, you want to provide an explicit mode.

The only reason we use `mode: preserve` in https://github.com/linux-system-roles/template/blob/master/tests/tests_include_vars_from_parent.yml#L13 is because it is a test file and the actual mode doesn't really matter.  For non-test code (i.e. production code), the mode really does matter.

Reply:
I was also wondering about it. But `/etc/rsyslog.d`, `rsyslog.conf`, and `/var/lib/rsyslog` are in the rsyslog rpm package. And the logging role ensures the package is installed. So, I thought we are supposed to use `preserve` there. One question is on `/var/lib/rsyslog`, not like the other two, the path is configurable. If a customer happens to change the path and the mode is `preserve` and the directory did not exist, the mode will be the default one.

Reply:
ok - then I guess `preserve` is ok here.
re: `/var/lib/rsyslog`
- does the rpm create this by default?
- if the user changes this value, is the user responsible for creating the directory and configuring the ownership/permission?

Reply:
Ahhh, sorry. Never mind. It's not programmable in the logging role level. I misunderstood by the var name not starting with `__` and it's defined in defaults/main.yml instead of vars/main.yml as the other "hardcoded" paths are...

If ok, may I move rename `rsyslog_work_dir` to `__rsyslog_work_dir` and move the definition to vars? Please note that `rsyslog_work_dir` does not appear any README files.

Reply:
> Ahhh, sorry. Never mind. It's not programmable in the logging role level. I misunderstood by the var name not starting with `__` and it's defined in defaults/main.yml instead of vars/main.yml as the other "hardcoded" paths are...
> 
> If ok, may I move rename `rsyslog_work_dir` to `__rsyslog_work_dir` and move the definition to vars? Please note that `rsyslog_work_dir` does not appear any README files.

yes.  Even though `rsyslog_work_dir` is defined in `roles/rsyslog/defaults/main.yml`, that isn't considered to be part of the public interface of the `logging` role.

Reply:
Done!
(and *shame shame*, the mode key in the `file` module does not take `preserve`... So, put the value back to the ones defined in the rpm spec files.)